SmartDeviceLink/SDLH264VideoEncoder.m

```diff
@@ -60,59 +60,18 @@ - (instancetype)initWithProtocol:(SDLVideoStreamingProtocol)protocol dimensions:
     _currentFrameNumber = 0;
     _videoEncoderSettings = properties;
     _dimensions = dimensions;
-    
     _delegate = delegate;
-    
-    OSStatus status;
-    
-    // Create a compression session
-    status = VTCompressionSessionCreate(NULL, (int32_t)dimensions.width, (int32_t)dimensions.height, kCMVideoCodecType_H264, NULL, self.sdl_pixelBufferOptions, NULL, &sdl_videoEncoderOutputCallback, (__bridge void *)self, &_compressionSession);
-    
-    if (status != noErr) {
-        if (error != NULL) {
-            *error = [NSError errorWithDomain:SDLErrorDomainVideoEncoder code:SDLVideoEncoderErrorConfigurationCompressionSessionCreationFailure userInfo:@{@"OSStatus":@(status), NSLocalizedDescriptionKey:@"Compression session could not be created"}];
-        }
-        
+
+    if (![self sdl_createCompressionSessionWithError:error]) {
         return nil;
     }
-    
-    CFRelease(_sdl_pixelBufferOptions);
-    _sdl_pixelBufferOptions = nil;
-    
-    // Validate that the video encoder properties are valid.
-    CFDictionaryRef supportedProperties;
-    status = VTSessionCopySupportedPropertyDictionary(self.compressionSession, &supportedProperties);
-    if (status != noErr) {
-        if (error != NULL) {
-            *error = [NSError errorWithDomain:SDLErrorDomainVideoEncoder code:SDLVideoEncoderErrorConfigurationCompressionSessionSetPropertyFailure userInfo:@{@"OSStatus":@(status), NSLocalizedDescriptionKey:[NSString stringWithFormat:@"\"%@\" are not supported properties.", supportedProperties]}];
-        }
-        
+
+    if (![self sdl_validateVideoEncoderPropertiesOnCompressionSession:self.compressionSession withError:error]) {
```

Comment:
The video encoder properties only need to be validated once. We could do all three steps at once, create the compression session, validate the video encoder properties and set the video encoder properties but when the compression session needs to be recreated, we only need to set the video encoder properties since they have already been validated.